dispatch/dispatch_types.go

```diff
@@ -72,7 +72,7 @@ type Mux struct {
 // Pipe defines an outbound object to the desired routine
 type Pipe struct {
 	// Channel to get all our lovely information
-	C <-chan interface{}
+	C chan interface{}
```

Comment:
Could un-export the field `c` just for protection reasons, below is a diff for changes. Let me know what you think. You can adjust the names if you want. 

```diff
diff --git a/dispatch/dispatch_test.go b/dispatch/dispatch_test.go
index d9e5fa664..6711bced1 100644
--- a/dispatch/dispatch_test.go
+++ b/dispatch/dispatch_test.go
@@ -426,7 +426,7 @@ func TestMux(t *testing.T) {
                        return
                }
                errChan <- nil
-       }(pipe.C, errChan, &wg)
+       }(pipe.c, errChan, &wg)

        wg.Wait()

@@ -506,7 +506,7 @@ func TestMuxPublish(t *testing.T) {
                }
        }(mux)

-       <-pipe.C
+       <-pipe.Wait()

        // Shut down dispatch system
        err = d.stop()
diff --git a/dispatch/dispatch_types.go b/dispatch/dispatch_types.go
index a4b149d6d..68091d108 100644
--- a/dispatch/dispatch_types.go
+++ b/dispatch/dispatch_types.go
@@ -72,9 +72,14 @@ type Mux struct {
 // Pipe defines an outbound object to the desired routine
 type Pipe struct {
        // Channel to get all our lovely information
-       C chan interface{}
+       c chan interface{}
        // ID to tracked system
        id uuid.UUID
        // Reference to multiplexer
        m *Mux
 }
+
+// Wait returns a channel to wait on for data
+func (p *Pipe) Wait() <-chan interface{} {
+       return p.c
+}
diff --git a/dispatch/mux.go b/dispatch/mux.go
index ecd983069..c597d0797 100644
--- a/dispatch/mux.go
+++ b/dispatch/mux.go
@@ -38,7 +38,7 @@ func (m *Mux) Subscribe(id uuid.UUID) (Pipe, error) {
                return Pipe{}, err
        }

-       return Pipe{C: ch, id: id, m: m}, nil
+       return Pipe{c: ch, id: id, m: m}, nil
 }

 // Unsubscribe returns channel to the pool for the full signature set
@@ -83,5 +83,5 @@ func (m *Mux) GetID() (uuid.UUID, error) {

 // Release returns the channel to the communications pool to be reused
 func (p *Pipe) Release() error {
-       return p.m.Unsubscribe(p.id, p.C)
+       return p.m.Unsubscribe(p.id, p.c)
 }
diff --git a/engine/rpcserver.go b/engine/rpcserver.go
index 0b51de192..769a1d0ea 100644
--- a/engine/rpcserver.go
+++ b/engine/rpcserver.go
@@ -714,7 +714,7 @@ func (s *RPCServer) GetAccountInfoStream(r *gctrpc.GetAccountInfoRequest, stream
        }()

        for {
-               data, ok := <-pipe.C
+               data, ok := <-pipe.Wait()
                if !ok {
                        return errDispatchSystem
                }
@@ -2188,7 +2188,7 @@ func (s *RPCServer) GetExchangeOrderbookStream(r *gctrpc.GetExchangeOrderbookStr
        }()

        for {
-               data, ok := <-pipe.C
+               data, ok := <-pipe.Wait()
                if !ok {
                        return errDispatchSystem
                }
@@ -2273,7 +2273,7 @@ func (s *RPCServer) GetTickerStream(r *gctrpc.GetTickerStreamRequest, stream gct
        }()

        for {
-               data, ok := <-pipe.C
+               data, ok := <-pipe.Wait()
                if !ok {
                        return errDispatchSystem
                }
@@ -2326,7 +2326,7 @@ func (s *RPCServer) GetExchangeTickerStream(r *gctrpc.GetExchangeTickerStreamReq
        }()

        for {
-               data, ok := <-pipe.C
+               data, ok := <-pipe.Wait()
                if !ok {
                        return errDispatchSystem
                }
diff --git a/exchanges/account/account_test.go b/exchanges/account/account_test.go
index 90fe5980a..8ba7cfc2d 100644
--- a/exchanges/account/account_test.go
+++ b/exchanges/account/account_test.go
@@ -213,7 +213,7 @@ func TestGetHoldings(t *testing.T) {
                for i := 0; i < 2; i++ {
                        c := time.NewTimer(time.Second)
                        select {
-                       case <-p.C:
+                       case <-p.Wait():
                        case <-c.C:
                        }
                }
```

Reply:
OK! I renamed Wait() to Channel() as it's more of a getter than a waiter.